File: unittest/scripts/py_devapi/scripts/mysqlx_farm_no_interactive.py
# Assumptions: ensure_schema_does_not_exist is available
# Assumes __uripwd is defined as <user>:<pwd>@<host>:<plugin_port>
# validateMemer and validateNotMember are defined on the setup script

dba.get_farm({"enforce":True})

#@ Farm: validating members
farm = dba.create_farm('devFarm', 'testing')

all_members = dir(farm)

# Remove the python built in members
members = []
for member in all_members:
  if not member.startswith('__'):
    members.append(member)

print "Farm Members: %d" % len(members)
validateMember(members, 'name')
validateMember(members, 'get_name')
validateMember(members, 'admin_type')
validateMember(members, 'get_admin_type')
validateMember(members, 'add_seed_instance')
validateMember(members, 'add_instance')
validateMember(members, 'remove_instance')
validateMember(members, 'get_replica_set')

#@ Farm: add_seed_instance
# Added this to enable add_instance, full testing of add_seed_instance is needed
farm.add_seed_instance({"host": '192.168.1.1'})

#@# Farm: add_instance errors
farm.add_instance()
farm.add_instance(5,6)
farm.add_instance(5)
farm.add_instance({"host": '192.168.1.1', "schema": 'abs'})
farm.add_instance({"host": '192.168.1.1', "user": 'abs'})
farm.add_instance({"host": '192.168.1.1', "password": 'abs'})
farm.add_instance({"host": '192.168.1.1', "authMethod": 'abs'})
farm.add_instance({"port": 33060})
farm.add_instance('')

#@# Farm: add_instance
farm.add_instance('192.168.1.1:33060')
farm.add_instance({"host": '192.168.1.1', "port": 1234})

# Cleanup
dba.drop_farm('devFarm', {"dropDefaultReplicaSet": True})

File: unittest/scripts/js_devapi/scripts/mysqlx_farm_no_interactive.js
// Assumptions: ensure_schema_does_not_exist is available
// Assumes __uripwd is defined as <user>:<pwd>@<host>:<plugin_port>
// validateMemer and validateNotMember are defined on the setup script
dba.dropMetadataSchema({enforce:true});

//@ Farm: validating members
var farm = dba.createFarm('devFarm', 'testing');

var members = dir(farm);

print("Farm Members:", members.length);
validateMember(members, 'name');
validateMember(members, 'getName');
validateMember(members, 'adminType');
validateMember(members, 'getAdminType');
validateMember(members, 'addSeedInstance');
validateMember(members, 'addInstance');
validateMember(members, 'removeInstance');
validateMember(members, 'getReplicaSet');

//@ Farm: addSeedInstance
// Added this to enable addInstance, full testing of addSeedInstance is needed
farm.addSeedInstance(__host_port);

//@# Farm: addInstance errors
farm.addInstance();
farm.addInstance(5,6);
farm.addInstance(5);
farm.addInstance({host: '192.168.1.1', schema: 'abs'});
farm.addInstance({host: '192.168.1.1', user: 'abs'});
farm.addInstance({host: '192.168.1.1', password: 'abs'});
farm.addInstance({host: '192.168.1.1', authMethod: 'abs'});
farm.addInstance({port: 33060});
farm.addInstance('');

//@# Farm: addInstance
farm.addInstance(__host_port);
farm.addInstance({host: '192.168.1.1', port: 1234});

// Cleanup
dba.dropFarm('devFarm', {dropDefaultReplicaSet: true});
